lambdas/build-stac/utils/stac.py

```diff
@@ -189,77 +191,106 @@ def generate_geometry_from_cmr(cmr_json, item) -> dict:
+            file.suffix, asset_media_type.get(file.suffix[1:], None)
+        )
     else:
         return asset_media_type
 
 
-def gen_asset(role: str, link: dict, item: dict) -> pystac.Asset:
-    if item.test_links and "http" in link.get("href"):
+def _roles(link: str, asset_roles: Union[list, dict], default: List[str]) -> List[str]:
+    if isinstance(asset_roles, dict):
+        file = Path(link)
+        return asset_roles.get(file.suffix, asset_roles.get(file.suffix[1:], default))
+    else:
+        return asset_roles
+
+
+def generate_asset(
+    roles: Union[str, Dict[str, List[str]]], link: dict, item: dict
+) -> pystac.Asset:
+    href = link.get("href")
+    if item.test_links and "http" in href:
         try:
-            requests.head(link.get("href"))
+            requests.head(href)
         except Exception as e:
             print(f"Caught error for link {link}: {e}")
             return None
-    asset_media_type = link.get("type")
-    # Fallback to asset_media_type defined in the item
-    if asset_media_type == None and role == "data":
-        asset_media_type = item.asset_media_type
 
-    return pystac.Asset(
-        roles=[role], href=link.get("href"), media_type=asset_media_type
+    # If type is in CMR link{} use that, else use the type from the asset_media_type
+    asset_media_type = link.get("type", _content_type(href, item.asset_media_type))
+    asset_roles = _roles(href, item.asset_roles, ["data"])
+
+    return pystac.Asset(roles=asset_roles, href=href, media_type=asset_media_type)
+
+
+def generate_link(
+    rel: Union[str, pystac.RelType],
+    target: Union[str, pystac.STACObject],
+    media_type: Optional[str] = None,
+    title: Optional[str] = None,
+    extra_fields: Optional[Dict[str, Any]] = None,
+) -> pystac.Link:
+    """
+    Generate a link object from a rel, target, media type, title, and extra fields.
+    """
+    return pystac.Link(
+        rel=rel,
+        target=target,
+        media_type=media_type,
+        title=title,
+        extra_fields=extra_fields,
     )
 
 
-def get_assets_from_cmr(cmr_json, item) -> dict[pystac.Asset]:
+def from_cmr_links(cmr_links, item) -> Tuple[List, Dict[str, pystac.Asset]]:
     """
     Generates a dictionary of pystac.Asset's from cmr_json links
-    TODO(aimee): This is using some heuristics and could probably be refined according to some standard in the future.
     """
     assets = {}
-    links = cmr_json["links"]
-    for link in links:
-        if link["rel"] == "http://esipfed.org/ns/fedsearch/1.1/data#":
+    links = []
+    for link in cmr_links:
+        if link["rel"].endswith("data#"):
             extension = os.path.splitext(link["href"])[-1].replace(".", "")
             if extension == "prj":
-                asset = gen_asset("metadata", link, item)
-                if asset:
-                    assets["metadata"] = asset
-            asset = gen_asset("data", link, item)
+                links.append(
```

Comment:
Looking at results for `prj file geospatial` in google led me to https://en.wikipedia.org/wiki/Shapefile - reading that page leads me to believe prj should actually be an asset, @wildintellect would you agree?

It seems like [prj files may be commonly used by ESRI applications](https://vsp.pnnl.gov/help/vsample/ESRI_PRJ_File.htm). And looking at that reference, I see that prj files are used to store metadata about a projection system.

Reply:
Right, these would most likely be assets with metadata roles - guess I got a little carried away 😅